contracts/token/WithdrawTokens.sol

```diff
@@ -0,0 +1,45 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC20/IERC20Upgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC721/IERC721Upgradeable.sol";
+
+/**
+ * @title Making a token withdrawable by the owner to a specified onlyOwner setable address
+ *
+ * @dev
+ */
+abstract contract WithdrawTokens is OwnableUpgradeable {
```

Comment:
issue: permissions is a problem.

The OZ AccessControl be used in some cases and Onwable in others. abstract the permissions needs abstracting (see OZ UPPS authorize)

Reply:
@CjHare 
Can you explain more?
Remove OwnableUpgradeable and instead inherit AccessControlUpgradeable : https://github.com/windranger-io/windranger-treasury/blob/main/contracts/bond/BondFactory.sol#L4 

Use the same roles/setup as BondFactory with Roles lib?

Should these contracts be upgradeable or no?